samples/subsys/fs/nffs/src/main.c

```diff
@@ -0,0 +1,149 @@
+#include <stdio.h>
+#include <stddef.h>
+#include <string.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <errno.h>
+#include <misc/printk.h>
+#include <misc/byteorder.h>
+#include <fs.h>
+#include <nrf.h>
+#include <device.h>
+#include <gpio.h>
+#include <board.h>
+
+int lock=0;
+
+struct device *button_device[1];
+
+static struct k_work button_work;
+
+static void button_pressed(struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	k_work_submit(&button_work);
+}
+
+void press(struct k_work *work)
+{
+	char buf[20];
+	int err=0;
+
+	fs_file_t fp;
+
+	err = fs_open(&fp,"/0.txt");
+	if (err != 0) {
+		printk("error %d while opening for read\n\r", err);
+	}
+
+	err = fs_read(&fp, buf, sizeof(buf));
+	if (err < 0) {
+		printk("error %d while reading\n\r", err);
+	}
+
+	printk("Data read from file is --> %s \n", buf);
```

Comment:
fs_read doesn't null terminate its output. You need to use the return value to just print the bytes which were read.